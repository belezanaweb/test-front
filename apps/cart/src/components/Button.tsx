import { HTMLAttributes } from "react";
import { tv, type VariantProps } from "tailwind-variants";

const styles = tv({
  base: "h-12 px-2 py-4 rounded font-bold leading-none text-base w-full flex justify-center items-center",
  variants: {
    color: {
      primary: "bg-purple-600 text-white",
      secondary: "bg-black text-white",
    },
  },
  defaultVariants: {
    color: "primary",
  },
});

type ButtonVariants = VariantProps<typeof styles>;
export type ButtonProps = HTMLAttributes<HTMLButtonElement> & ButtonVariants;

export function Button({ className, color, ...props }: ButtonProps) {
  return (
    <button
      className={styles({
        className,
        color,
      })}
      type="button"
      {...props}
    />
  );
}
