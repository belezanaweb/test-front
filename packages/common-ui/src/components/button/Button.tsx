'use client'

import { VariantProps } from 'class-variance-authority'
import clsx from 'clsx'
import { ButtonHTMLAttributes } from 'react'
import { buttonStyles } from './styles'

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & VariantProps<typeof buttonStyles>

export function Button({ intent, className, ...props }: ButtonProps) {
  return (
    <button
      className={clsx(
        buttonStyles({
          intent,
          disabled: props.disabled
        }),
        className
      )}
      disabled={props.disabled}
      {...props}
    />
  )
}
