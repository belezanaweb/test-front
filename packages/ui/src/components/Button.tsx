'use client'

export const Button = () => {
  return <button onClick={() => alert('boop')}>Button Component</button>
}
