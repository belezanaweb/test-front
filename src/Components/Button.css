.btn {
  font-family: "Helvetica Neue", "Helvetica", "Arial", "sans-serif";
  font-weight: 600;
  font-size: calc(16px + 1vmin);

  text-transform: uppercase;

  background: #ff6a00;
  color: #ffffff;

  width: 100%;
  height: 60px;

  border: 0;
  border-radius: 4px;

  box-shadow: #d45a00 0 3px 0, 1px 5px 3px rgba(35, 26, 26, 0.25);

  margin-top: 10px;
  margin-bottom: 10px;
}

.btn:hover {
  background: rgb(212, 90, 0);
  box-shadow: unset;
  border: unset;
}
