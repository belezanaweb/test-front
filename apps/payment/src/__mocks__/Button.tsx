import { ButtonProps } from 'cart/Button';

export default function Button({ children, ...props }: ButtonProps) {
  return (
    <button type="button" {...props}>
      {children}
    </button>
  );
}
