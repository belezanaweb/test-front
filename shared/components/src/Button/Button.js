import { useFormikContext } from "formik";
import * as S from "./Button.styles";

export const Button = ({ label, ...props }) => {
  const { isSubmitting } = useFormikContext();

  return (
    <S.Button type="submit" {...props} disabled={isSubmitting}>
      {label}
    </S.Button>
  );
};
