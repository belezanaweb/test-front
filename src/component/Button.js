import React from 'react'
import * as S from './Button.styles'

const Button = ({ text, onClick, ...props }) => {
  return (
    <S.StyledButton onClick={onClick} {...props}>
      {text}
    </S.StyledButton>
  )
}

export default Button
