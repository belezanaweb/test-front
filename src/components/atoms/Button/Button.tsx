import * as Styles from './Button.styles'
interface ButtonProps {
  text: string;
  type: ('button' | 'submit');
  onClick: () => void;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ text, disabled = false, type, onClick }) => {
  return (
    <Styles.Container>
      <Styles.Button type={type} disabled={disabled} onClick={onClick}>
        {text}
      </Styles.Button>
    </Styles.Container>

  )
}

export default Button
