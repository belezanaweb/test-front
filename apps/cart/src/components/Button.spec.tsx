import { render, fireEvent } from '@testing-library/react';

import Button from './Button';

describe('Button', () => {
  it('should render the button correctly with default props', () => {
    const onClick = jest.fn();
    const { container, getByText } = render(
      <Button onClick={onClick}>Click me</Button>,
    );
    const button = getByText('Click me');

    expect(container.firstChild).toMatchSnapshot();
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('bg-purple-600', 'text-white');
    expect(button).not.toHaveClass('bg-black', 'text-white');
    expect(button).not.toHaveClass(
      'bg-gray-300',
      'text-gray-500',
      'cursor-not-allowed',
    );
    expect(button).not.toBeDisabled();
    expect(onClick).not.toHaveBeenCalled();
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should render the button correctly with custom props', () => {
    const onClick = jest.fn();
    const { container, getByText } = render(
      <Button color="secondary" onClick={onClick}>
        Click me
      </Button>,
    );
    const button = getByText('Click me');

    expect(container.firstChild).toMatchSnapshot();
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('bg-black', 'text-white');
    expect(button).not.toHaveClass('bg-purple-600', 'text-white');
    expect(button).not.toHaveClass(
      'bg-gray-300',
      'text-gray-500',
      'cursor-not-allowed',
    );
    expect(button).not.toBeDisabled();
    expect(onClick).not.toHaveBeenCalled();
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should render the button correctly with disabled prop', () => {
    const onClick = jest.fn();
    const { container, getByText } = render(
      <Button disabled onClick={onClick}>
        Click me
      </Button>,
    );
    const button = getByText('Click me');

    expect(container.firstChild).toMatchSnapshot();
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass(
      'bg-gray-300',
      'text-gray-500',
      'cursor-not-allowed',
    );
    expect(button).not.toHaveClass('bg-purple-600', 'text-white');
    expect(button).not.toHaveClass('bg-black', 'text-white');
    expect(button).toBeDisabled();
    expect(onClick).not.toHaveBeenCalled();
    fireEvent.click(button);
    expect(onClick).not.toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  it('should call the onClick callback when clicked', () => {
    const onClick = jest.fn();
    const { getByText } = render(<Button onClick={onClick}>Click me</Button>);
    const button = getByText('Click me');

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
