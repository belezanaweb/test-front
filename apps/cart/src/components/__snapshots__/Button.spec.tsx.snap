// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button correctly with custom props 1`] = `
<button
  class="h-12 px-2 py-4 rounded font-bold text-base w-full flex justify-center items-center bg-black text-white"
  type="button"
>
  Click me
</button>
`;

exports[`Button should render the button correctly with default props 1`] = `
<button
  class="h-12 px-2 py-4 rounded font-bold text-base w-full flex justify-center items-center bg-purple-600 text-white"
  type="button"
>
  Click me
</button>
`;

exports[`Button should render the button correctly with disabled prop 1`] = `
<button
  class="h-12 px-2 py-4 rounded font-bold text-base w-full flex justify-center items-center bg-gray-300 text-gray-500 cursor-not-allowed"
  disabled=""
  type="button"
>
  Click me
</button>
`;
